tests/test_addons.py

```diff
@@ -442,6 +477,178 @@ def test_add_file_osfstorage_log(self):
         assert_equal(self.node.logs.count(), nlogs + 1)
         assert('urls' in self.node.logs.filter(action='osf_storage_file_added')[0].params)
 
+    def test_add_log_updates_cache_create(self):
+        self.configure_osf_addon()
+        path = 'lollipop'
+        url = self.node.api_url_for('create_waterbutler_log')
+        payload = self.build_payload(metadata={
+            'materialized': path,
+            'kind': 'file',
+            'path': path,
+            'size': 100,
+            'nid': self.node._id,
+        })
+        self.app.put_json(url, payload, headers={'Content-Type': 'application/json'})
+        self.node.reload()
+        assert self.node.storage_usage == 100
+
+    def test_add_log_updates_cache_update(self):
+        self.configure_osf_addon()
+        path = 'lollipop'
+        url = self.node.api_url_for('create_waterbutler_log')
+        payload = self.build_payload(metadata={
+            'materialized': path,
+            'kind': 'file',
+            'path': path,
+            'size': 120,
+            'nid': self.node._id,
+        }, action='update')
+        self.app.put_json(url, payload, headers={'Content-Type': 'application/json'})
+        self.node.reload()
+        assert self.node.storage_usage == 120
+
+    def test_add_log_updates_cache_move(self):
+        ''' Renames are actually just move functions with the same nid for src and dest '''
+        self.configure_osf_addon()
+        url = self.node.api_url_for('create_waterbutler_log')
+        self.file2.create_version(self.user, {
+            'object': '06d80e',
+            'service': 'cloud',
+            osfstorage_settings.WATERBUTLER_RESOURCE: 'osf',
+        }, {
+            'size': 250,
+            'contentType': 'img/png'
+        }).save()
+
+        assert self.node.storage_usage == 250
+
+        payload = self.build_payload_with_dest(
+            action='move',
+            source={
+                'materialized': 'lollipop',
+                'kind': 'file',
+                'nid': self.node._id,
+                'provider': 'osfstorage',
+                'name': 'new.txt',
+                'path': '/lollipop'
+            },
+            destination={
+                'path': '/lollipop',
+                'materialized': 'lollipop',
+                'kind': 'file',
+                'provider': 'osfstorage',
+                'nid': self.node2._id,
+                'name': 'new.txt',
+            },
+        )
+        self.app.put_json(url, payload, headers={'Content-Type': 'application/json'})
+
+        key = cache_settings.STORAGE_USAGE_KEY.format(target_id=self.node._id)
+        assert storage_usage_cache.get(key) == 0
+
+        self.node2.reload()
+        assert self.node2.storage_usage == 250
+
+    def test_add_log_updates_cache_copy(self):
```

Comment:
Add a test of copy within the same node?

Reply:
👍